src/main/java/com/google/finapp/SpannerDaoJDBCImpl.java

```diff
@@ -0,0 +1,197 @@
+        ps.executeUpdate();
+        System.out.printf("New role created: %s\n", roleName);
+      }
+    }
+  }
+
+
+  public void moveAccountBalance(
+      ByteArray fromAccountId, ByteArray toAccountId, BigDecimal amount)
+      throws SQLException {
+    try (Connection connection = DriverManager.getConnection(this.connectionUrl)) {
+      // begin transaction
+      connection.setAutoCommit(false);
+      try (
+          PreparedStatement readStatement = connection.prepareStatement(
+              "SELECT AccountId, Balance FROM Account WHERE (AccountId = ? or AccountId = ?)"
+          );
+          PreparedStatement updateAccountStatement = connection.prepareStatement(
+              "UPDATE Account SET Balance = ? WHERE AccountId = ?"
+          );
+          PreparedStatement insertTransactionStatement = connection.prepareStatement(
+              "INSERT INTO TransactionHistory (AccountId, Amount, IsCredit, EventTimestamp)"
+                  + "VALUES (?, ?, ?, PENDING_COMMIT_TIMESTAMP()), "
+                  + "(?, ?, ?, PENDING_COMMIT_TIMESTAMP())")) {
+        byte[] fromAccountIdArray = fromAccountId.toByteArray();
+        byte[] toAccountIdArray = toAccountId.toByteArray();
+        BigDecimal[] accountBalances = readAccountBalances(
+            fromAccountIdArray, toAccountIdArray, readStatement);
+        updateAccount(fromAccountIdArray,
+            accountBalances[0].subtract(amount),
+            updateAccountStatement);
+        updateAccount(toAccountIdArray,
+            accountBalances[1].add(amount), updateAccountStatement);
+        updateAccountStatement.executeBatch();
+        insertTransaction(
+            fromAccountIdArray, toAccountIdArray, amount,
+            insertTransactionStatement);
+        insertTransactionStatement.executeUpdate();
+        connection.commit();
+        System.out.printf("Balance of %s moved.\n", amount.toString());
+      }
+    }
+  }
+
+  void getAccountMetadata(ByteArray accountId) throws SpannerException {
+
+  }
+
+
+  private BigDecimal[] readAccountBalances(
+      byte[] fromAccountId, byte[] toAccountId, PreparedStatement preparedStatement)
+      throws SQLException {
+    // preparedStatement: "SELECT AccountId, Balance FROM Account WHERE (AccountId = ? or AccountId = ?)"
+    preparedStatement.setBytes(1, fromAccountId);
+    preparedStatement.setBytes(2, toAccountId);
+    java.sql.ResultSet resultSet = preparedStatement.executeQuery();
+    BigDecimal[] results = new BigDecimal[2];
+    while (resultSet.next()) {
+      byte[] currentId = resultSet.getBytes("AccountId");
+      if (Arrays.equals(currentId, fromAccountId)) {
+        results[0] = resultSet.getBigDecimal("Balance");
+      } else {
+        results[1] = resultSet.getBigDecimal("Balance");
+      }
+    }
+    return results;
+  }
+
+
+  private void updateAccount(byte[] accountId, BigDecimal newBalance,
+      PreparedStatement preparedStatement) throws SQLException {
```

Comment:
This formatting doesn't seem to be from google-java-format. Can you please run that ?